sdk/storage/storage-file-share/src/ShareFileClient.ts

```diff
@@ -1395,31 +1396,36 @@ export class ShareFileClient extends StorageClient {
   // High Level functions
 
   /**
-   * ONLY AVAILABLE IN BROWSERS.
-   *
-   * Uploads a browser Blob/File/ArrayBuffer/ArrayBufferView object to an Azure File.
+   * Uploads a Buffer(Node)/Blob/ArrayBuffer/ArrayBufferView to an Azure File.
    *
-   * @param {Blob | ArrayBuffer | ArrayBufferView} browserData Blob, File, ArrayBuffer or ArrayBufferView
+   * @param {Buffer | Blob | ArrayBuffer | ArrayBufferView} data Buffer(Node), Blob, ArrayBuffer or ArrayBufferView
    * @param {FileParallelUploadOptions} [options]
    * @returns {Promise<void>}
    */
-  public async uploadBrowserData(
-    browserData: Blob | ArrayBuffer | ArrayBufferView,
+  public async uploadData(
+    data: Buffer | Blob | ArrayBuffer | ArrayBufferView,
     options: FileParallelUploadOptions = {}
   ): Promise<void> {
-    const { span, spanOptions } = createSpan(
-      "ShareFileClient-uploadBrowserData",
-      options.tracingOptions
-    );
+    const { span, spanOptions } = createSpan("ShareFileClient-uploadData", options.tracingOptions);
     try {
-      const browserBlob = new Blob([browserData]);
-      return this.uploadSeekableBlob(
-        (offset: number, size: number): Blob => {
-          return browserBlob.slice(offset, offset + size);
-        },
-        browserBlob.size,
-        { ...options, tracingOptions: { ...options!.tracingOptions, spanOptions } }
-      );
+      if (isNode && data instanceof Buffer) {
+        const bufferToStream = new ReadableStreamBuffer();
+        bufferToStream.put(data);
```

Comment:
Where is `ReadableStreamBuffer` from? Does it have enough efficiency for `bufferToStream.put(data);` when data size is large?

Reply:
It's from a new dependency "stream-buffers" - https://github.com/Azure/azure-sdk-for-js/pull/6060#discussion_r342883069

I haven't looked at its limits. I can verify, probably add some tests, throw errors accordingly.

Reply:
Since we're receiving a buffer that contains all the users data, stream-buffers' implementation should be fairly efficient - it will grow its internal buffer once to fit the incoming buffer, and copy the data over. Unfortunately for large buffers, this copy operation might be expensive. It is also probably unnecessary for our use case where we aren't adding multiple buffers but have all the data up front. It would be nice if we could just hand the buffer to buffer-stream to use as its internal buffer. IIUC, we are using this primarily for its ability to take a large buffer and chunk it up and send it over a stream. I wonder if it's worth just implementing this ourselves in a utility rather than depending on the library? It would likely be more efficient.

Reply:
Thanks for the feedback! I'll think of other ways. :)

Reply:
To avoid the extra copy operation of the buffer by "stream-buffers".
- Removed the dependency on `stream-buffers`
- Came up with a new `uploadBuffer`(private method) which is similar to `uploadResetableStream` except for the callback function. Difference is shown here - https://github.com/Azure/azure-sdk-for-js/pull/6060#discussion_r344024632